src/utils/googleTagManager.ts

```diff
@@ -155,14 +166,63 @@ export const trackEvent = ({
   action: string
   label?: EventLabel
 }): void => {
-  const dataLayer = {
+  const dataLayer: EventDataLayer = {
     event,
     chainId: _getChainId(),
     eventCategory: category,
     eventAction: action,
     eventLabel: tryParse(label),
   }
 
+  track(dataLayer)
+}
+
+const ETHEREUM_ADDRESS_REGEX = /0x[a-fA-F0-9]{40}/g
+const ETHEREUM_ADDRESS = 'ETHEREUM_ADDRESS'
+
+type SafeAppEventDataLayer = {
+  event: GTM_EVENT
```

Comment:
From what it seems, this is always `GTM_EVENT.SAFE_APP`. I would set the `event` type to only accept that.